### ISSUE TITLE
Docs: 422 has wrong description in example and wrong reference link

### ISSUE THREAD

**Issue by miguno** [confluentinc/kafka-rest#63]
http://confluent.io/docs/current/kafka-rest/docs/api.html#errors

```json
HTTP/1.1 422 OK
Content-Type: application/vnd.kafka.v1+json

{
    "error_code": 422,
    "message": "records may not be empty"
}
```

I think the first line should read:

```json
HTTP/1.1 422 Unprocessable Entity
```

Also, a bit further, down there is a link: [422 Unprocessable Entity](http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.4.23) which points to RFC2616.  This RFC does not describe 422.  The last 4xx in the document is 417.

**Comment by ewencp** [comment]
Good catch. I just pushed 2d823f42da to fix the status message which was a trivial fix.

The link is actually generated by a sphinx-contrib's httpdomain code. They had a number of exceptions for other codes, but not the WebDAV ones. I submitted an pull request to fix that here
https://bitbucket.org/birkenfeld/sphinx-contrib/pull-request/92 but we'll need to wait for it to be merged and for 1.3.1 to be released, then update `docs/requirements.txt` to use at least that version. So I'll leave this open until those last couple of pieces can be resolved.
